deepmd/jax/model/base_model.py

```
@@ -91,17 +91,65 @@ def eval_output(
+                        # atomic_ret[_kk]: [nf, nloc, *def]
+                        atomic_ret = self.atomic_model.forward_common_atomic(
+                            cc_ext[None, ...],
+                            extended_atype[None, ...],
+                            nlist[None, ...],
+                            mapping=mapping[None, ...] if mapping is not None else None,
+                            fparam=fparam[None, ...] if fparam is not None else None,
+                            aparam=aparam[None, ...] if aparam is not None else None,
+                        )
+                        nloc = nlist.shape[0]
+                        cc_loc = jax.lax.stop_gradient(cc_ext)[:nloc, ...]
+                        cc_loc = jnp.reshape(cc_loc, [nloc, *[1] * def_ndim, 3])
```

Ruff

Function definition does not bind loop variable `def_ndim`

(B023)